# Spurs Data Analysis

### I loaded the necessary packages

In [445]:
import pandas as pd
import numpy as np
import csv
import os

### I imported all the csvs from the <b>"raw_season_csvs"</b> folder and combined them into one dataframe
It appears to all be correct (each season should have 380 total games, so one game per row results in 380 rows per file).

<br><i>The data comes from [football-data.co.uk](https://www.football-data.co.uk/englandm.php)</i>
<br><i>The below cell was written with the help of Claude Code</i>

In [446]:
csv_dir = "/Users/hgorledeenn/Desktop/CJS/0126data_studio/project-1-spurs/raw_season_csvs"

csv_files = sorted([f for f in os.listdir(csv_dir) if f.endswith('.csv')])
all_rows = []
all_columns = set()

for file in csv_files:
    filepath = os.path.join(csv_dir, file)
    row_count = 0
    with open(filepath, 'r', encoding='latin-1') as f:
        reader = csv.DictReader(f)
        for row in reader:
            row['season'] = file.split('.')[0].split('-')[1]
            all_rows.append(row)
            all_columns.update(row.keys())
            row_count += 1
    print(f"Loaded {file}: {row_count} rows")

combined_df = pd.DataFrame(all_rows)

print(f"\nCombined dataframe shape: {combined_df.shape}")
print(f"Number of columns: {len(combined_df.columns)}")


Loaded 10-0203.csv: 380 rows
Loaded 11-0304.csv: 380 rows
Loaded 12-0405.csv: 380 rows
Loaded 13-0506.csv: 380 rows
Loaded 14-0607.csv: 380 rows
Loaded 15-0708.csv: 380 rows
Loaded 16-0809.csv: 380 rows
Loaded 17-0910.csv: 380 rows
Loaded 18-1011.csv: 380 rows
Loaded 19-1112.csv: 380 rows
Loaded 20-1213.csv: 380 rows
Loaded 21-1314.csv: 380 rows
Loaded 22-1415.csv: 381 rows
Loaded 23-1516.csv: 380 rows
Loaded 24-1617.csv: 380 rows
Loaded 25-1718.csv: 380 rows
Loaded 26-1819.csv: 380 rows
Loaded 27-1920.csv: 380 rows
Loaded 28-2021.csv: 380 rows
Loaded 29-2122.csv: 380 rows
Loaded 30-2223.csv: 380 rows
Loaded 31-2324.csv: 380 rows
Loaded 32-2425.csv: 380 rows
Loaded 33-2526.csv: 260 rows
Loaded 8-0001.csv: 380 rows
Loaded 9-0102.csv: 380 rows

Combined dataframe shape: (9761, 218)
Number of columns: 218


### I selected only the columns I need for my analysis and saved the pared down dataframe as a csv

In [447]:
all_games = combined_df[['season','Date', 'HomeTeam','AwayTeam','FTHG','FTAG','FTR','HTHG','HTAG','HTR','HS','AS','HST','AST']]

In [448]:
all_games.to_csv('output_csvs/all_games.csv', index=False)

In [449]:
all_games.head()

season        Date   HomeTeam    AwayTeam FTHG FTAG FTR HTHG HTAG HTR  HS  \
0   0203  17/08/2002  Blackburn  Sunderland    0    0   D    0    0   D  15   
1   0203  17/08/2002   Charlton     Chelsea    2    3   A    2    1   H   5   
2   0203  17/08/2002    Everton   Tottenham    2    2   D    1    0   H  13   
3   0203  17/08/2002     Fulham      Bolton    4    1   H    3    1   H  13   
4   0203  17/08/2002      Leeds    Man City    3    0   H    2    0   H  13   

   AS HST AST  
0   7   5   3  
1  21   5  12  
2  10   9   5  
3   3   6   1  
4  18   8  10

### Filter to only games where Tottenham is the home or away team, reset the index, standardize the date column

In [450]:
only_tot = all_games[(all_games['HomeTeam'] == 'Tottenham') | (all_games['AwayTeam'] == 'Tottenham')]
only_tot = only_tot.reset_index(drop=True)
only_tot['Date'] = pd.to_datetime(only_tot['Date'], dayfirst=True, format='mixed')
only_tot.head()

season       Date   HomeTeam     AwayTeam FTHG FTAG FTR HTHG HTAG HTR  HS  \
0   0203 2002-08-17    Everton    Tottenham    2    2   D    1    0   H  13   
1   0203 2002-08-24  Tottenham  Aston Villa    1    0   H    1    0   H   8   
2   0203 2002-08-27   Charlton    Tottenham    0    1   A    0    1   A  10   
3   0203 2002-08-31  Tottenham  Southampton    2    1   H    1    1   D  18   
4   0203 2002-09-11     Fulham    Tottenham    3    2   H    0    2   A  13   

   AS HST AST  
0  10   9   5  
1  18   3   8  
2   7   5   1  
3  15   8   8  
4   7   8   4

### Make a blank dataframe for my tottenham-centric stats

In [451]:
tottenham = pd.DataFrame()

### Defining a few functions for use below

In [452]:
def ft_results(row):
    if row['TFTG'] > row['OTFTG']:
        return 'W'
    elif row['TFTG'] < row['OTFTG']:
        return 'L'
    else:
        return 'D'
    
def ht_results(row):
    if row['THTG'] > row['OTHTG']:
        return 'W'
    elif row['THTG'] < row['OTHTG']:
        return 'L'
    else:
        return 'D'

### Below, I populate my new dataframe with the same stats as in ```only_tot```, just from Tottenham's perspective (mostly just using np.where to determine whether to pull in the home team stat or the away team stat)

#### Data Diary for my new columns:
<b>Season</b>: Season of the game (in the format 0203, where the season spanned 2002 and 2003)
<br><b>Date</b>: Date of the game
<br><b>OT</b>: The team Tottenham played
<br><b>TL</b>: The location of the game, from Tottenham's perspective (H = Home, A = Away)
<br><b>TFTG</b>: Tottenham Full-Time Goals
<br><b>OTFTG</b>: Other Team Full-Time Goals
<br><b>FTR</b>: Full Time Result (W if Tottenham win, L if Tottenham loss, D if draw)
<br><b>THTG</b>: Tottenham Half-Time Goals
<br><b>OTHTG</b>: Other Team Half-Time Goals
<br><b>HTR</b>: Half Time Result (W if Tottenham winning, L if Tottenham losing, D if tied)
<br><b>TS</b>: Total shots by Tottenham
<br><b>OTS</b>: Total shots by the Other Team
<br><b>TST</b>: Tottenham shots on target
<br><b>OTST</b>: Other Team shots on target
<br><b>FTGD</b>: Full-Time goal differential (+ if Tottenham won, - if Tottenham lost, 0 if draw)
<br><b>HTGD</b>: Half-Time goal differential (+ if Tottenham won, - if Tottenham lost, 0 if draw)
<br><b>OTST</b>: The last name of Tottenham's head coach for that game
<br><b>Coach</b>: The last name of Tottenham's head coach for that game
<br><b>Coach</b>: The last name of Tottenham's head coach for that game
<br><b>Coach</b>: The last name of Tottenham's head coach for that game
<br><b>Coach</b>: The last name of Tottenham's head coach for that game
<br><b>Coach</b>: The last name of Tottenham's head coach for that game
<br><b>Coach</b>: The last name of Tottenham's head coach for that game

In [453]:
tottenham['Season'] = only_tot['season']
tottenham['Date'] = only_tot['Date']
tottenham['OT'] = np.where(only_tot['HomeTeam'] == 'Tottenham', only_tot['AwayTeam'], only_tot['HomeTeam'])
tottenham['TL'] = np.where(only_tot['HomeTeam'] == 'Tottenham', "H", "A")
tottenham['TFTG'] = np.where(only_tot['HomeTeam'] == 'Tottenham', only_tot['FTHG'], only_tot['FTAG'])
tottenham['OTFTG'] = np.where(only_tot['HomeTeam'] == 'Tottenham', only_tot['FTAG'], only_tot['FTHG'])
tottenham['FTR'] = tottenham.apply(ft_results, axis=1)
tottenham['THTG'] = np.where(only_tot['HomeTeam'] == 'Tottenham', only_tot['HTHG'], only_tot['HTAG'])
tottenham['OTHTG'] = np.where(only_tot['HomeTeam'] == 'Tottenham', only_tot['HTAG'], only_tot['HTHG'])
tottenham['HTR'] = tottenham.apply(ht_results, axis=1)
tottenham['TS'] = np.where(only_tot['HomeTeam'] == 'Tottenham', only_tot['HS'], only_tot['AS'])
tottenham['OTS'] = np.where(only_tot['HomeTeam'] == 'Tottenham', only_tot['AS'], only_tot['HS'])
tottenham['TST'] = np.where(only_tot['HomeTeam'] == 'Tottenham', only_tot['HST'], only_tot['AST'])
tottenham['OTST'] = np.where(only_tot['HomeTeam'] == 'Tottenham', only_tot['AST'], only_tot['HST'])


In [454]:
numeric_columns = ['TFTG', 'OTFTG', 'THTG', 'OTHTG', 'TS', 'OTS', 'TST', 'OTST']

for col in numeric_columns:
    tottenham[col] = pd.to_numeric(tottenham[col], errors='coerce')

tottenham.dtypes

Season            object
Date      datetime64[ns]
OT                object
TL                object
TFTG               int64
OTFTG              int64
FTR               object
THTG               int64
OTHTG              int64
HTR               object
TS                 int64
OTS                int64
TST                int64
OTST               int64
dtype: object

In [455]:
tottenham['FTGD'] = tottenham['TFTG'] - tottenham['OTFTG']
tottenham['HTGD'] = tottenham['THTG'] - tottenham['OTHTG']


## Next, I created a dataframe with each Tottenham coach and their start and end dates in the time period my data covers

In [456]:
import pandas as pd

coaches = [
    {"coach":"George Graham","start":"1998-10-01","end":"2001-03-16"},
    {"coach":"Interim Manager*","start":"2001-03-17","end":"2001-04-01"},
    {"coach":"Glen Hoddle","start":"2001-04-02","end":"2003-09-21"},
    {"coach":"Interim Manager*","start":"2003-09-22","end":"2004-06-03"},
    {"coach":"Jacques Santini","start":"2004-06-04","end":"2004-11-05"},
    {"coach":"Martin Jol","start":"2004-11-06","end":"2007-10-26"},
    {"coach":"Interim Manager*","start":"2007-10-27","end":"2007-10-29"},
    {"coach":"Juande Ramos","start":"2007-10-30","end":"2008-10-25"},
    {"coach":"Harry Redknapp","start":"2008-10-26","end":"2012-06-15"},
    {"coach":"André Villas-Boas","start":"2012-06-16","end":"2013-12-16"},
    {"coach":"Tim Sherwood","start":"2013-12-17","end":"2014-05-13"},
    {"coach":"Mauricio Pochettino","start":"2014-05-14","end":"2019-11-19"},
    {"coach":"José Mourinho","start":"2019-11-20","end":"2021-04-19"},
    {"coach":"Interim Manager*","start":"2021-04-20","end":"2021-06-29"},
    {"coach":"Nuno Espírito Santo","start":"2021-06-30","end":"2021-11-01"},
    {"coach":"Antonio Conte","start":"2021-11-02","end":"2023-03-26"},
    {"coach":"Interim Manager*","start":"2023-03-27","end":"2023-06-05"},
    {"coach":"Ange Postecoglou","start":"2023-06-06","end":"2025-06-06"},
    {"coach":"Thomas Frank","start":"2025-06-07","end":"2026-02-11"},
    {"coach":"Igor Tudor","start":"2026-02-12","end":"2026-03-01"}
]

coach_df = pd.DataFrame(coaches)
coach_df["start"] = pd.to_datetime(coach_df["start"])
coach_df["end"]   = pd.to_datetime(coach_df["end"])


In [457]:
tottenham = tottenham.sort_values("Date")
coach_df = coach_df.sort_values("start")

tottenham = pd.merge_asof(tottenham, coach_df, left_on="Date", right_on="start", direction="backward")
tottenham = tottenham[tottenham["Date"] <= tottenham["end"]]

tottenham

Season       Date             OT TL  TFTG  OTFTG FTR  THTG  OTHTG HTR  TS  \
0     0001 2000-08-19        Ipswich  H     3      1   W     2      1   W  20   
1     0001 2000-08-22  Middlesbrough  A     1      1   D     1      0   W  11   
2     0001 2000-08-26      Newcastle  A     0      2   L     0      1   L  10   
3     0001 2000-09-05        Everton  H     3      2   W     1      2   L  11   
4     0001 2000-09-11       West Ham  H     1      0   W     0      0   D  15   
..     ...        ...            ... ..   ...    ...  ..   ...    ...  ..  ..   
971   2526 2026-01-17       West Ham  H     1      2   L     0      1   L  21   
972   2526 2026-01-24        Burnley  A     2      2   D     1      1   D  18   
973   2526 2026-02-01       Man City  H     2      2   D     0      2   L  12   
974   2526 2026-02-07     Man United  A     0      2   L     0      1   L   7   
975   2526 2026-02-10      Newcastle  H     1      2   L     0      1   L  16   

     OTS  TST  OTST  FTGD  HTGD          coach      start        end  
0     15    6     5     2     1  George Graham 1998-10-01 2001-03-16  
1     12    4     6     0     1  George Graham 1998-10-01 2001-03-16  
2     15    2     6    -2    -1  George Graham 1998-10-01 2001-03-16  
3     11    7     6     1    -1  George Graham 1998-10-01 2001-03-16  
4     11    7     5     1     0  George Graham 1998-10-01 2001-03-16  
..   ...  ...   ...   ...   ...            ...        ...        ...  
971   17    6     4    -1    -1   Thomas Frank 2025-06-07 2026-02-11  
972    9   11     4     0     0   Thomas Frank 2025-06-07 2026-02-11  
973   15    6     3     0    -2   Thomas Frank 2025-06-07 2026-02-11  
974   23    1    10    -2    -1   Thomas Frank 2025-06-07 2026-02-11  
975   21    2     5    -1    -1   Thomas Frank 2025-06-07 2026-02-11  

[976 rows x 19 columns]

In [458]:
tottenham = tottenham.drop(columns=['start', 'end'])

In [459]:
tottenham.head()

Season       Date             OT TL  TFTG  OTFTG FTR  THTG  OTHTG HTR  TS  \
0   0001 2000-08-19        Ipswich  H     3      1   W     2      1   W  20   
1   0001 2000-08-22  Middlesbrough  A     1      1   D     1      0   W  11   
2   0001 2000-08-26      Newcastle  A     0      2   L     0      1   L  10   
3   0001 2000-09-05        Everton  H     3      2   W     1      2   L  11   
4   0001 2000-09-11       West Ham  H     1      0   W     0      0   D  15   

   OTS  TST  OTST  FTGD  HTGD          coach  
0   15    6     5     2     1  George Graham  
1   12    4     6     0     1  George Graham  
2   15    2     6    -2    -1  George Graham  
3   11    7     6     1    -1  George Graham  
4   11    7     5     1     0  George Graham

### I calcualted some rolling averages for certain stats

In [460]:
tottenham['FTGDRA'] = tottenham['FTGD'].rolling(window=5).mean()
tottenham['HTGDRA'] = tottenham['HTGD'].rolling(window=5).mean()
tottenham['TFTGRA'] = tottenham['TFTG'].rolling(window=5).mean()
tottenham['OTFTGRA'] = tottenham['OTFTG'].rolling(window=5).mean()
tottenham['TSTRA'] = tottenham['TST'].rolling(window=5).mean()
tottenham['OTSTRA'] = tottenham['OTST'].rolling(window=5).mean()

In [461]:
tottenham.head(15)

Season       Date             OT TL  TFTG  OTFTG FTR  THTG  OTHTG HTR  ...  \
0    0001 2000-08-19        Ipswich  H     3      1   W     2      1   W  ...   
1    0001 2000-08-22  Middlesbrough  A     1      1   D     1      0   W  ...   
2    0001 2000-08-26      Newcastle  A     0      2   L     0      1   L  ...   
3    0001 2000-09-05        Everton  H     3      2   W     1      2   L  ...   
4    0001 2000-09-11       West Ham  H     1      0   W     0      0   D  ...   
5    0001 2000-09-16       Charlton  A     0      1   L     0      1   L  ...   
6    0001 2000-09-23       Man City  H     0      0   D     0      0   D  ...   
7    0001 2000-09-30          Leeds  A     3      4   L     1      0   W  ...   
8    0001 2000-10-14       Coventry  A     1      2   L     0      2   L  ...   
9    0001 2000-10-21          Derby  H     3      1   W     2      1   W  ...   
10   0001 2000-10-28        Chelsea  A     0      3   L     0      2   L  ...   
11   0001 2000-11-04     Sunderland  H     2      1   W     1      0   W  ...   
12   0001 2000-11-11    Aston Villa  A     0      2   L     0      1   L  ...   
13   0001 2000-11-19      Liverpool  H     2      1   W     2      1   W  ...   
14   0001 2000-11-25      Leicester  H     3      0   W     2      0   W  ...   

    OTST  FTGD  HTGD          coach  FTGDRA  HTGDRA TFTGRA  OTFTGRA  TSTRA  \
0      5     2     1  George Graham     NaN     NaN    NaN      NaN    NaN   
1      6     0     1  George Graham     NaN     NaN    NaN      NaN    NaN   
2      6    -2    -1  George Graham     NaN     NaN    NaN      NaN    NaN   
3      6     1    -1  George Graham     NaN     NaN    NaN      NaN    NaN   
4      5     1     0  George Graham     0.4     0.0    1.6      1.2    5.2   
5      5    -1    -1  George Graham    -0.2    -0.4    1.0      1.2    5.0   
6      2     0     0  George Graham    -0.2    -0.6    0.8      1.0    5.2   
7      7    -1     1  George Graham     0.0    -0.2    1.4      1.4    5.8   
8      5    -1    -2  George Graham    -0.4    -0.4    1.0      1.4    5.4   
9      7     2     1  George Graham    -0.2    -0.2    1.4      1.6    5.6   
10     7    -3    -2  George Graham    -0.6    -0.4    1.4      2.0    5.2   
11     3     1     1  George Graham    -0.4    -0.2    1.8      2.2    5.4   
12     6    -2    -1  George Graham    -0.6    -0.6    1.2      1.8    4.8   
13     4     1     1  George Graham    -0.2     0.0    1.4      1.6    4.6   
14     5     3     2  George Graham     0.0     0.2    1.4      1.4    4.4   

    OTSTRA  
0      NaN  
1      NaN  
2      NaN  
3      NaN  
4      5.6  
5      5.6  
6      4.8  
7      5.0  
8      4.8  
9      5.2  
10     5.6  
11     5.8  
12     5.6  
13     5.4  
14     5.0  

[15 rows x 23 columns]

In [462]:
tottenham['FTGD'].mean()

np.float64(0.33811475409836067)

In [463]:
tottenham.to_csv('output_csvs/tottenham_centric.csv', index=False, encoding="utf-8-sig")

In [464]:
tottenham_frank_only = tottenham[(tottenham['coach'] == 'Thomas Frank')]
tottenham_frank_only.head()

Season       Date           OT TL  TFTG  OTFTG FTR  THTG  OTHTG HTR  ...  \
950   2526 2025-08-16      Burnley  H     3      0   W     1      0   W  ...   
951   2526 2025-08-23     Man City  A     2      0   W     2      0   W  ...   
952   2526 2025-08-30  Bournemouth  H     0      1   L     0      1   L  ...   
953   2526 2025-09-13     West Ham  A     3      0   W     0      0   D  ...   
954   2526 2025-09-20     Brighton  A     2      2   D     1      2   L  ...   

     OTST  FTGD  HTGD         coach  FTGDRA  HTGDRA TFTGRA  OTFTGRA  TSTRA  \
950     4     3     1  Thomas Frank    -0.8     0.2    1.0      1.8    2.4   
951     4     2     2  Thomas Frank    -0.4     0.6    1.2      1.6    3.0   
952     6    -1    -1  Thomas Frank    -0.2     0.6    1.2      1.4    3.0   
953     4     3     0  Thomas Frank     0.8     0.6    1.8      1.0    3.8   
954     4     0    -1  Thomas Frank     1.4     0.2    2.0      0.6    4.0   

     OTSTRA  
950     6.2  
951     6.6  
952     5.8  
953     5.2  
954     4.4  

[5 rows x 23 columns]

In [465]:
tottenham['2NDHTG'] = tottenham['TFTG'] - tottenham['THTG']

In [466]:
tottenham.head()

Season       Date             OT TL  TFTG  OTFTG FTR  THTG  OTHTG HTR  ...  \
0   0001 2000-08-19        Ipswich  H     3      1   W     2      1   W  ...   
1   0001 2000-08-22  Middlesbrough  A     1      1   D     1      0   W  ...   
2   0001 2000-08-26      Newcastle  A     0      2   L     0      1   L  ...   
3   0001 2000-09-05        Everton  H     3      2   W     1      2   L  ...   
4   0001 2000-09-11       West Ham  H     1      0   W     0      0   D  ...   

   FTGD  HTGD          coach  FTGDRA  HTGDRA  TFTGRA OTFTGRA  TSTRA  OTSTRA  \
0     2     1  George Graham     NaN     NaN     NaN     NaN    NaN     NaN   
1     0     1  George Graham     NaN     NaN     NaN     NaN    NaN     NaN   
2    -2    -1  George Graham     NaN     NaN     NaN     NaN    NaN     NaN   
3     1    -1  George Graham     NaN     NaN     NaN     NaN    NaN     NaN   
4     1     0  George Graham     0.4     0.0     1.6     1.2    5.2     5.6   

   2NDHTG  
0       1  
1       0  
2       0  
3       2  
4       1  

[5 rows x 24 columns]

In [467]:
tottenham['TST'].mean()

np.float64(6.170081967213115)

In [468]:
tottenham[tottenham['coach']=='Thomas Frank']['TST'].mean()

np.float64(3.769230769230769)

In [469]:
tottenham_frank_only.to_csv('output_csvs/frank_only.csv', index=False, encoding="utf-8-sig")

In [470]:
tottenham_by_coach = pd.DataFrame()

In [471]:
tottenham_by_coach = (
    tottenham
    .groupby("coach")
    .agg(
        ft_goals_for=('TFTG', 'sum'),
        ft_goals_against=('OTFTG','sum'),
        ft_avg_goals=('TFTG', 'mean'),
        ht_goals_for=('THTG', 'sum'),
        ht_goals_against=('OTHTG','sum'),
        ht_avg_goals=('THTG', 'mean'),
        overall_ft_goal_diff=('FTGD', 'sum'),
        avg_ft_goal_diff=('FTGD', 'mean'),
        overall_ht_goal_diff=('HTGD','sum'),
        avg_ht_goal_diff=('HTGD', 'mean'),
        games=("coach", "count"),
        wins=("FTR", lambda x: (x == "W").sum()),
        losses=("FTR", lambda x: (x == "L").sum()),
        draws=("FTR", lambda x: (x == "D").sum())
    )
    .reset_index()
)

In [472]:
tottenham_by_coach['overall_goal_diff'] = tottenham_by_coach['ft_goals_for'] - tottenham_by_coach['ft_goals_against']
tottenham_by_coach['win_pct'] = tottenham_by_coach['wins'] / tottenham_by_coach['games']
tottenham_by_coach['loss_pct'] = tottenham_by_coach['losses'] / tottenham_by_coach['games']
tottenham_by_coach['draw_pct'] = tottenham_by_coach['draws'] / tottenham_by_coach['games']


In [473]:
tottenham_by_coach.head(15)

coach  ft_goals_for  ft_goals_against  ft_avg_goals  \
0     André Villas-Boas            81                67      1.500000   
1      Ange Postecoglou           138               126      1.815789   
2         Antonio Conte           112                64      2.000000   
3         George Graham            33                38      1.137931   
4           Glen Hoddle           116               141      1.303371   
5        Harry Redknapp           228               161      1.583333   
6      Interim Manager*            78                80      1.529412   
7       Jacques Santini             6                 8      0.545455   
8         José Mourinho            97                67      1.672414   
9          Juande Ramos            53                50      1.514286   
10           Martin Jol           168               146      1.486726   
11  Mauricio Pochettino           372               206      1.841584   
12  Nuno Espírito Santo             9                16      0.900000   
13         Thomas Frank            36                37      1.384615   
14         Tim Sherwood            40                30      1.818182   

    ht_goals_for  ht_goals_against  ht_avg_goals  overall_ft_goal_diff  \
0             27                27      0.500000                    14   
1             59                63      0.776316                    12   
2             45                30      0.803571                    48   
3             19                21      0.655172                    -5   
4             51                52      0.573034                   -25   
5            103                64      0.715278                    67   
6             30                37      0.588235                    -2   
7              3                 5      0.272727                    -2   
8             53                26      0.913793                    30   
9             21                16      0.600000                     3   
10            72                71      0.637168                    22   
11           163                91      0.806931                   166   
12             6                 5      0.600000                    -7   
13            14                20      0.538462                    -1   
14            15                16      0.681818                    10   

    avg_ft_goal_diff  overall_ht_goal_diff  avg_ht_goal_diff  games  wins  \
0           0.259259                     0          0.000000     54    29   
1           0.157895                    -4         -0.052632     76    31   
2           0.857143                    15          0.267857     56    32   
3          -0.172414                    -2         -0.068966     29     9   
4          -0.280899                    -1         -0.011236     89    32   
5           0.465278                    39          0.270833    144    71   
6          -0.039216                    -7         -0.137255     51    20   
7          -0.181818                    -2         -0.181818     11     3   
8           0.517241                    27          0.465517     58    27   
9           0.085714                     5          0.142857     35    10   
10          0.194690                     1          0.008850    113    47   
11          0.821782                    72          0.356436    202   113   
12         -0.700000                     1          0.100000     10     5   
13         -0.038462                    -6         -0.230769     26     7   
14          0.454545                    -1         -0.045455     22    13   

    losses  draws  overall_goal_diff   win_pct  loss_pct  draw_pct  
0       13     12                 14  0.537037  0.240741  0.222222  
1       34     11                 12  0.407895  0.447368  0.144737  
2       15      9                 48  0.571429  0.267857  0.160714  
3       11      9                 -5  0.310345  0.379310  0.310345  
4       39     18                -25  0.359551  0.438202  0.202247  
5       36     37

In [474]:
tottenham_by_coach.to_csv('output_csvs/by_coach_summary.csv', index=False, encoding="utf-8-sig")

In [475]:
tot_by_coach_wldonly = tottenham_by_coach[['coach','win_pct', 'draw_pct', 'loss_pct']].copy()
tot_by_coach_wldonly.head()

coach   win_pct  draw_pct  loss_pct
0  André Villas-Boas  0.537037  0.222222  0.240741
1   Ange Postecoglou  0.407895  0.144737  0.447368
2      Antonio Conte  0.571429  0.160714  0.267857
3      George Graham  0.310345  0.310345  0.379310
4        Glen Hoddle  0.359551  0.202247  0.438202

In [476]:
tot_by_coach_wldonly.to_csv('output_csvs/by_coach_wldonly.csv', index=False, encoding="utf-8-sig")

In [477]:
htr_vs_ftr = pd.crosstab(tottenham_frank_only['FTR'], tottenham_frank_only['HTR'])
FH_vs_SH_goals = pd.crosstab(tottenham['2NDHTG'], tottenham['THTG'])

In [478]:
FH_vs_SH_goals.head()

THTG      0    1   2   3  4  5
2NDHTG                        
0       212  137  50   8  2  1
1       165  116  39  14  1  0
2        87   51  16   8  1  0
3        21   22   8   1  0  0
4         5    6   2   1  0  0

In [479]:
FH_vs_SH_goals.to_csv('output_csvs/fh_vs_sh_gls_all.csv', index=True)

In [480]:
htr_vs_ftr.head()

HTR  D  L  W
FTR         
D    4  3  1
L    2  9  0
W    2  0  5

In [481]:
htr_vs_ftr.to_csv('output_csvs/heatmapdata_frankonly.csv', index=True)